Gulp pack should not run on dataplane packages

KarishmaGhiya opened the pull request:
The gulp-pack step in the management plane pipelines were running on data plane packages too, aometimes causing our PRs to fail. So we want to update the gulp-packstep to not include the data plane packages - they should run only on management plane packages

ramya-rao-a commented:
@KarishmaGhiya Is this PR still applicable? Can you add more details in the PR description?

qiaozha commented:
@KarishmaGhiya  I think you might forget to change https://github.com/Azure/azure-sdk-for-js/blob/master/package.json#L31 here. such as add a `"prebuild": "your data-plane list generation script"`. In this way,  when you run `npm build` it will automatically run prebuild first.

KarishmaGhiya commented:
> @KarishmaGhiya I think you might forget to change https://github.com/Azure/azure-sdk-for-js/blob/master/package.json#L31 here. such as add a `"prebuild": "your data-plane list generation script"`. In this way, when you run `npm build` it will automatically run prebuild first.

That sounds good! Thanks!!

KarishmaGhiya commented:
@qiaozha There are 2 places where I added the step back in the yaml files, since we don't run build before the `gulp pack` step in the ci yaml file and before the `npm run build` step in the pr yaml file. Can you check these?

weshaggard commented on .scripts/common.ts:
```diff
@@ -111,18 +111,15 @@ function isPackageFolderPath(folderPath: string, packagesToIgnore: string[]): bo
-  "@azure/template",
-  "testhub"
-];
+export const dataplaneListFilename = "list-dataplane.json";
```
Why generate a file and then read it here? Why not just the package list like you are doing in the script that generates the data-plane list?

KarishmaGhiya replied:
I changed the solution from generating a file to call a TS script which has a function return both the list of package names and list of package's folder names. That way I can even skip the paths to the samples of the dataplane packages.

qiaozha commented:
> @qiaozha There are 2 places where I added the step back in the yaml files, since we don't run build before the gulp pack step in the ci yaml file and before the npm run build step in the pr yaml file. Can you check these?

Should we change ci.yaml or pr yaml `gulp pack` step into `npm run build`? that would be more consistent.

KarishmaGhiya commented:
@qiaozha  @weshaggard I changed the solution from generating a file to call a TS script which has a function return both the list of package names and list of package's folder names. That way I can even skip the paths to the samples of the dataplane packages.
Also, I removed the pre-build step from the package.json as this solution doesn't require it anymore. @qiaozha  for your credScan tool PR, you can write reuse the variables from common.ts script to skip the "packagesToIgnore" and "foldersToIgnore"

weshaggard commented on package.json:
```diff
@@ -51,10 +51,10 @@
+    "prettier": "^1.16.4",
     "ts-node": "^7.0.1",
-    "typescript": "^3.2.2",
+    "typescript": "^3.7.4",
```
Did you intend to change the version of TS? Make sure that is what you want as I know different versions of TS have some compatibility issues.

KarishmaGhiya replied:
No I didn't intend to do this. I'll revert it back.

weshaggard commented on eng/pipelines/mgmt-ci.yml:
```diff
@@ -11,7 +11,6 @@ trigger:
       - sdk/keyvault/keyvault-certificates/
       - sdk/keyvault/keyvault-keys/
       - sdk/keyvault/keyvault-secrets/
-      - sdk/keyvault/keyvault/
```
Why was this removed?

KarishmaGhiya replied:
There is no such path hat currently exists in the repo

KarishmaGhiya commented on .scripts/generateDataplaneList.ts:
```diff
@@ -0,0 +1,26 @@
+  for (const proj of rushJson.projects) {
+    const filePath = path.join(baseDir, proj.projectFolder, "package.json");
+    packageFolders.push(path.basename(path.dirname(filePath)));
+    packageNames.push(proj.projectName);
```
sorry.. my bad! I'll submit a fix PR for this

qiaozha replied:
why don't you just reopen this one?
